src/Listener/ControllerAuthorizationListener.php

```diff
@@ -15,25 +20,48 @@ class ControllerAuthorizationListener
 
     private $templateParams = array();
 
-    private $permissions = array(
-        'ajax.editabsencereasons' => array(100),
-        'ajax.holidaydelete' => array(100),
-        'ajax.changepassword' => array(21),
-    );
+    private $permissions = array();
+
+    protected $entityManager;
 
-    public function __construct(\Twig_Environment $twig)
+    public function __construct(\Twig_Environment $twig, EntityManagerInterface $em)
     {
+        $this->permissions = Yaml::parseFile(__DIR__."/../../config/permissions.yaml");
+
         $this->twig = $twig;
 
         $this->templateParams = $GLOBALS['templates_params'];
         $this->droits = $GLOBALS['droits'];
+        $this->entityManager = $em;
     }
 
     public function onKernelRequest(GetResponseEvent $event)
     {
+        $page = $event->getRequest()->getPathInfo();
+        $page = preg_replace('/([a-z\/]*).*/', "$1", $page);
+        $page = rtrim($page, '/add');
+        $page = rtrim($page, '/');
+
+        // Droits necessair<es pour consulter la page en cours
+        $accesses = $this->entityManager->getRepository(Access::class)->findBy(array('page' => $page));
+        $logged_in = $this->entityManager->find(Agent::class, $_SESSION['login_id']);
+
+        $authorized = true;
         $route = $event->getRequest()->attributes->get('_route');
 
-        if (!$this->canAccess($route)) {
+        if(!in_array($route, $this->permissions)){
+            $authorized = $logged_in ? $logged_in->can_access($accesses) : false;
+        }
+
+        if ($_SESSION['oups']["Auth-Mode"] == 'Anonyme' ) {
+            foreach ($accesses as $access) {
+                if ($access->groupe_id() == '99') {
+                    $authorized = true;
+                }
+            }
+        }
+
+        if (!($this->canAccess($route) and $authorized == 'true')) {
```

Comment:
$authorized == 'true' and $authorized == true are differents.
Here $authorized will *never* equal to the string 'true'.

Anyway, we don't need this check if we early return in the condition line 52.